checked/ref.go

```diff
@@ -169,3 +172,73 @@ func (c *RefCount) DecWrites() {
 func (c *RefCount) NumWriters() int {
 	return int(atomic.LoadInt32(&c.writes))
 }
+
+// TrackObject sets up the initial internal state of the Ref for
+// leak detection.
+func (c *RefCount) TrackObject(v interface{}) {
+	if atomic.LoadUint64(&_LeakDetectionFlag) == 0 {
+		return
+	}
+
+	var size int
+
+	switch v := reflect.ValueOf(v); v.Kind() {
+	case reflect.Ptr:
+		size = int(v.Type().Elem().Size())
+	case reflect.Array, reflect.Slice, reflect.Chan:
+		size = int(v.Type().Elem().Size()) * v.Cap()
+	case reflect.String:
+		size = v.Len()
+	default:
+		size = int(v.Type().Size())
+	}
+
+	runtime.SetFinalizer(c, func(c *RefCount) {
+		if c.NumRef() == 0 {
+			return
+		}
+
+		origin := getDebuggerRef(c).String()
+
+		_Leaks.Lock()
+		// Keep track of bytes leaked, not objects.
+		_Leaks.M[origin] += uint64(size)
+		_Leaks.Unlock()
+	})
+}
+
+// EnableLeakDetection turns leak detection on.
+func EnableLeakDetection() {
+	atomic.StoreUint64(&_LeakDetectionFlag, 1)
+}
+
+// DisableLeakDetection turns leak detection off.
+func DisableLeakDetection() {
```

Comment:
Could we make these methods similar to the other enable/disable methods in `debug.go`? i.e.

```go
func SetTraceLeak(value bool) {
  traceLeak = value
}
```